core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -6411,7 +6411,11 @@ public final String[] listUsers(String filter, int maxItemLimit) throws UserStor
             // Using the short-circuit. User name comes with the domain name.
             String domain = filter.substring(0, index);
 
-            UserStoreManager secManager = getSecondaryUserStoreManager(domain);
+            UserStoreManager secManager = this;
+            if (StringUtils.isNotEmpty(domain) && !StringUtils.equals(getMyDomainName(), domain)) {
```

Comment:
```suggestion
            shall we use StringUtils.equalsIgnoreCase(getMyDomainName(), domain))?
```

Reply:
we can use. Updated with [0b507ae](https://github.com/wso2/carbon-kernel/pull/3804/commits/0b507aec5150f16645043bdd7ae13a82a1c34a08)